NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java

```diff
@@ -187,6 +207,105 @@ public GroupPermission getPermissionforGroup(Group group){
 		return permhandle.getGroupPermission(group);
 	}
 	
+	// == GROUP INTERACTIONS ============================================================== //
+	
+	public List<UUID> getMembers(String groupname) {
+		Group group = getGroup(groupname);
+		if (group == null) { 
+			return Lists.newArrayList();
+		}
+		return group.getAllMembers();
+	}
+	
+	public List<UUID> getMembers(String groupname, PlayerType rank) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return Lists.newArrayList();
+		}
+		return group.getAllMembers(rank);
+	}
+	
+	public PlayerType getMemberRank(String groupname, UUID player) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return null;
+		}
+		return group.getMember(player);
+	}
+	
+	public boolean isMember(String groupname, UUID player) {		
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return false;
+		}
+		return group.isMember(player);
+	}
+	
+	public boolean hasAccess(String groupname, UUID player, PermissionType perm) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return false;
+		}
+		
+		GroupPermission perms = getPermissionforGroup(group);
+		PlayerType rank = group.getMember(player);	
+		
+		return perms.isAccessible(rank, perm);
+	}
+	
+	/**
+	 * @param groupname - String name of the group
+	 * @return false if the group can be operated on, true otherwise
+	 */
+	public boolean isDisciplined(String groupname) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return true;
+		}
+		
+		return group.isDisciplined();
+	}
+	
+	public boolean areLinked(String supername, String subname) {
+		Group supergroup = getGroup(supername);
+		Group subgroup = getGroup(subname);
+		if (supergroup == null || subgroup == null) {
+			return false;
+		}
+		
+		return Group.areLinked(supergroup, subgroup);
+	}
+	
+	public boolean link(String supername, String subname) {
+		Group supergroup = getGroup(supername);
+		Group subgroup = getGroup(subname);
+		if (supergroup == null || subgroup == null) {
+			return false;
+		}
+		
+		boolean success = Group.link(supergroup, subgroup);
+		if (success) {
+			groupManagerDao.addSubGroup(supername, subname);
```

Comment:
I initially had these in just the Group class, but then I realized that the constructor was adding the subgroups to the database again while it was linking the group to them, although I'm not sure if that does anything. 

So to be safe I moved these to the GroupManager, but forgot that the stuff that called unlink in Group and needed to update the database. If it doesn't do anything, then putting these back into the Group class is probably best. Otherwise the stuff in Group will need to be moved to the GroupManager I guess... 

Is the checkRecacheGroup method the thing that causes shards to recache? I didn't put those into the link and unlink commands.